File: examples/functions/rest-api-methods/products/[id]/delete.php
<?php
// DELETE /products/:id — id arrives directly from [id] filename
function handler($event, $params) {
    $id = $params["id"] ?? "";
    return [
        "status" => 200,
        "headers" => ["Content-Type" => "application/json"],
        "body" => json_encode(["id" => (int)$id, "deleted" => true]),
    ];
}

File: examples/functions/rest-api-methods/products/[id]/delete.lua
-- DELETE /products/:id — id arrives directly from [id] filename
local cjson = require("cjson")

local function handler(event, params)
    local id = params.id or ""
    return {
        status = 200,
        headers = { ["Content-Type"] = "application/json" },
        body = cjson.encode({
            id = tonumber(id),
            deleted = true,
        }),
    }
end

return handler
